### PR TITLE
Make windows factory-reset sudoless

### PR DESCRIPTION
**Pull request by Nino-K**
The script utilizes a WSL call to trigger a command-line call inside the rancher-desktop distribution. This allows executing iptables flush and deletion commands within the actual rancher-desktop distribution. Previously, these commands were run in the test (ubuntu) distribution, which used a different iptables configuration than the one employed by the rancher desktop.

Furthermore, the script addresses the issue with the xargs -l option. This option is not available in busybox

Fixes: https://github.com/rancher-sandbox/rancher-desktop/issues/5126

### CODE REVIEW

File: bats/tests/helpers/commands.bash [hunk 1]
```diff
@@ -80,3 +80,6 @@ rdshell() {
 rdsudo() {
     rdshell sudo "$@"
 }
+wsl_exe() {
+    "$SYSTEM32/wsl.exe" "$@"
+}
```
Comment (jandubois) [inline bats/tests/helpers/commands.bash:85]: Let's call just `wsl.exe` and let the OS look it up in the `PATH`. We don't need a wrapper function for it.

We are using the full path for all utilities bundled with Rancher Desktop, but for OS utilities we assume they are on the `PATH`. Just like we use `cp` and don't write `/bin/cp`.

File: bats/tests/helpers/paths.bash [hunk 2]
```diff
@@ -100,6 +100,7 @@ if is_windows; then
     PATH_DISTRO="$PATH_DATA/distro"
     PATH_DISTRO_DATA="$PATH_DATA/distro-data"
     PATH_EXTENSIONS="$PATH_DATA/extensions"
+    SYSTEM32="/mnt/c/Windows/System32"
```
Comment (jandubois) [inline bats/tests/helpers/paths.bash:103]: We should not need this. If we did, we would have to translate it from `%SystemRoot%\System32`.

File: bats/tests/helpers/vm.bash [hunk 3]
```diff
@@ -17,6 +17,21 @@ pkill_by_path() {
     fi
 }
 
+windows_cleanup() {
+    # Fixed in WSL 0.64.0, however below is a workaround for windows10
+    # https://github.com/microsoft/WSL/releases/tag/0.64.0
+    export WSL_UTF8=1
+    WSLENV="$WSLENV":WSL_UTF8
```
Comment (jandubois) [inline bats/tests/helpers/vm.bash:24]: Please write it like this (I'll write up some style guidelines soon):

```suggestion
    WSLENV="${WSLENV}:WSL_UTF8"
```

File: bats/tests/helpers/vm.bash [hunk 4]
```diff
@@ -17,6 +17,21 @@ pkill_by_path() {
...
+    # match rancher-desktop and not rancher-desktop-data
+    if is_windows && wsl_exe -l -v | grep -Po '(?<=^| )rancher-desktop(?!-data(?=\b|$))'; then
```
Comment (jandubois) [inline bats/tests/helpers/vm.bash:26]: What is the reason for the complex condition? Just run all the commands inside the block with `run` so we ignore the error:

```suggestion
    if is_windows; then
```

Or if you really want to check if the distro exist, run a simple command in it:

```suggestion
    if is_windows && wsl.exe -d rancher-desktop true >/dev/null; then
```

**Review by jandubois — CHANGES_REQUESTED** [review]
I made a number of change requests, but I'm also wondering: if the commands need to be executed inside the `rancher-desktop` distro, why do we need them at all?

We are just going to uninstall the distro right afterwards. So all this code would have no effect anyways, unless it is global across distros, and then I don't understand why this PR is necessary.

Let's discuss in person.

**Comment by Nino-K** [comment]
> We are just going to uninstall the distro right afterwards. So all this code would have no effect anyways, unless it is global across distros, and then I don't understand why this PR is necessary.

The core of issue is that the iptables rules persist even after a factory reset, where the distro is completely removed. This is why we encounter intermittent networking issues. It could be related to the way WSL2 memory retention and caching work (although not confirmed), but the crucial point is that the iptables rule persists until you shut down the hyper-v kernel (wsl --shutdown), which is why things start working when we use 'wsl --shutdown' command.

One way to test this out is to create a simple rule:
e.g 
```
# check the chain first
iptables -t nat -L PREROUTING

# now assert the rule
iptables -t nat -A PREROUTING -d 192.168.127.2 -j DNAT --to-destination 127.0.0.1
```
Now perform a `rdctl factory-reset` and then start RD again.

```
# check the chain again
iptables -t nat -L PREROUTING
```
you will notice our rule has persisted

```
Chain PREROUTING (policy ACCEPT)
target     prot opt source               destination
KUBE-SERVICES  all  --  anywhere             anywhere             /* kubernetes service portals */
CNI-HOSTPORT-DNAT  all  --  anywhere             anywhere             ADDRTYPE match dst-type LOCAL
DNAT       all  --  anywhere             192.168.127.2        to:127.0.0.1
DOCKER     all  --  anywhere             anywhere             ADDRTYPE match dst-type LOCAL
```

Therefore, I believe manually flushing the iptables and removing the unwanted rules is still necessary to ensure there is no flakiness, as long as we flush the correct iptables. 

Previously, we flushed the iptables inside the test distro (Ubuntu), which made no sense at all because that iptables is not even used or populated with anything.

File: bats/tests/helpers/vm.bash [hunk 5]
```diff
@@ -17,6 +17,21 @@ pkill_by_path() {
     fi
 }
 
+windows_cleanup() {
+    # Fixed in WSL 0.64.0, however below is a workaround for windows10
+    # https://github.com/microsoft/WSL/releases/tag/0.64.0
+    export WSL_UTF8=1
+    WSLENV="${WSLENV}:WSL_UTF8"
```
Comment (jandubois) [inline bats/tests/helpers/vm.bash:24]: Why is this necessary? I suspect it should have been removed when we removed grepping the output of `wsl -l -v` for distro names,

File: bats/tests/helpers/vm.bash [hunk 6]
```diff
@@ -17,6 +17,21 @@ pkill_by_path() {
...
+    # match rancher-desktop and not rancher-desktop-data
```
Comment (jandubois) [inline bats/tests/helpers/vm.bash:25]: I think this comment is outdated and can be removed.

File: bats/tests/helpers/vm.bash [hunk 7]
```diff
@@ -17,6 +17,21 @@ pkill_by_path() {
     fi
 }
 
+windows_cleanup() {
```
Comment (jandubois) [inline bats/tests/helpers/vm.bash:20]: I think this should be named `wsl_cleanup`, but I would also be fine with leaving it inside the `factory_reset` function.

**Review by jandubois — CHANGES_REQUESTED** [review]
I think there is still vestigial code in the PR.

**Review by jandubois — APPROVED** [review]

**Pull request merged**